netmiko/base_connection.py

```diff
@@ -968,7 +969,7 @@ def send_config_set(self, config_commands=None, exit_config_mode=True, delay_fac
             raise ValueError("Invalid argument passed into send_config_set")
 
         # Send config commands
-        output = self.config_mode()
+        output = self.config_mode(config_mode_cmd) if config_mode_cmd else self.config_mode()
```

Comment:
Might be better to do this with a kwargs pattern:
```
kwargs = {}
if config_mode_cmd:
    kwargs['config_command'] = config_mode_cmd
output = self.config_mode(**kwargs)
```
Something similar could be achieved with positional args.